internal/dataplane/sendconfig/strategy.go

```diff
@@ -0,0 +1,58 @@
+package sendconfig
+
+import (
+	"context"
+
+	"github.com/kong/deck/dump"
+	"github.com/kong/deck/file"
+	"github.com/kong/go-kong/kong"
+)
+
+// UpdateStrategy is the way we approach updating data-plane's configuration, depending on its type.
+type UpdateStrategy interface {
+	// Update applies targetConfig to the data-plane.
+	Update(ctx context.Context, targetContent *file.Content) error
+
+	// MetricsProtocol returns a string describing the update strategy type to be used in metrics.
+	MetricsProtocol() string
+}
+
+type UpdateClient interface {
+	IsKonnect() bool
+	KonnectRuntimeGroup() string
+	AdminAPIClient() *kong.Client
+}
+
+// ResolveUpdateStrategy returns an UpdateStrategy based on the client and configuration.
+// The UpdateStrategy can be either UpdateStrategyDBMode or UpdateStrategyInMemory. Both
+// of them implement different ways to populate Kong instances with data-plane configuration.
+func ResolveUpdateStrategy(
+	client UpdateClient,
+	config Config,
+) UpdateStrategy {
+	adminAPIClient := client.AdminAPIClient()
+
+	if !config.InMemory || client.IsKonnect() {
```

Comment:
Konnect enforces db mode 🤔 ?

Reply:
Yup. In case the client's `IsKonnect()=true`, we're sure it should be db-mode. We could distinguish here an error case where `InMemory=true && IsKonnect()=true`, would you prefer it this way?